corgi/tasks/brew.py

```diff
@@ -96,6 +95,16 @@ def slow_fetch_brew_build(
         },
         completion_time=completion_dt,
     )
+    if created:
+        try:
+            relation = ProductComponentRelation.objects.get(
+                build_id=build_id, build_type=build_type
+            )
+        except ProductComponentRelation.DoesNotExist:
+            pass
+        else:
+            relation.software_build = softwarebuild
+            relation.save()
```

Comment:
Yes, there are often multiple relations for the same build. You question made me review the SoftwareBuild.get_product_details function used by SoftwareBuild.save_product_taxonomy and I think there is a bug there where if a build is associated with a Stream it will also be incorrectly be associated with all variants and channels in that stream family. We should stop doing that.
It's fine to assume that build to variant mapping also includes all parents in the hierarchy it's not correct to say that just because a build is associated with a stream, that it also can be associated with all the variant children of that stream. We should instead be relying on ERRATA_TOOL relations to determine if a build is associated with a variant or not. I think doing this will enable us to build SBOM per variant. SBOM per variant should allow:

- EL8/9 versions of SBOM
- Minor version SBOM (not just .z)